core/src/main/java/org/apache/gravitino/storage/relational/service/TableColumnMetaService.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.storage.relational.service;
+
+import com.google.common.collect.Lists;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.gravitino.meta.ColumnEntity;
+import org.apache.gravitino.meta.TableEntity;
+import org.apache.gravitino.storage.relational.mapper.TableColumnMapper;
+import org.apache.gravitino.storage.relational.po.ColumnPO;
+import org.apache.gravitino.storage.relational.po.TablePO;
+import org.apache.gravitino.storage.relational.utils.POConverters;
+import org.apache.gravitino.storage.relational.utils.SessionUtils;
+
+public class TableColumnMetaService {
+
+  private static final TableColumnMetaService INSTANCE = new TableColumnMetaService();
+
+  private TableColumnMetaService() {}
+
+  public static TableColumnMetaService getInstance() {
+    return INSTANCE;
+  }
+
+  public List<ColumnPO> getColumnsByTableIdAndVersion(Long tableId, Long version) {
+    List<ColumnPO> columnPOs =
+        SessionUtils.getWithoutCommit(
+            TableColumnMapper.class,
+            mapper -> mapper.listColumnPOsByTableIdAndVersion(tableId, version));
+
+    // Filter out the deleted columns
+    return columnPOs.stream()
+        .filter(c -> c.getColumnOpType() != ColumnPO.ColumnOpType.DELETE.value())
+        .collect(Collectors.toList());
+  }
+
+  public void insertColumnPOs(TablePO tablePO, List<ColumnEntity> columnEntities) {
```

Comment:
Please make this method `package` accessible, by the way, since the value will not do commit operation, I believe it's just for some special cases, could you add some description here.

Reply:
Just add some comments and change the scope of the methods.